lib/active_fedora/autosave_association.rb

```diff
@@ -243,10 +243,9 @@ def association_valid?(reflection, record)
 
         unless valid = record.valid?
           if reflection.options[:autosave]
-            record.errors.each do |attribute, message|
+            record.errors.messages.each do |attribute, messages|
```

Comment:
`#messages` returns a hash with attributes as keys and arrays of messages as values in rails 5.2 through 7.0.  This was the only way I could find that worked in all supported rails versions.